build-aws.json

```diff
@@ -18,8 +17,16 @@
          "instance_type":"t2.medium",
          "region":"ca-central-1",
          "secret_key":"{{user `aws_secret_access_key`}}",
-         "source_ami":"ami-b3d965d7",
-         "ssh_username":"ubuntu"
```

Comment:
Introducing new logic of source_ami_filter as explained here. Instead of hard-coding an ami ID, packer will perform a sort of look up function (most likely invoking AWS' APIs) in order to return a matching AMI. If not found, pakcer will naturally fail.